File: frontend/src/types/Statistics.ts
import {ScoreMap} from "./ScoreMap.ts";

export type Statistics = {
  longestWinningStreak: ScoreMap;
  longestLosingStreak: ScoreMap;
  gamesPlayed: ScoreMap;
  gamesWon: ScoreMap;
  averageTime: ScoreMap;
  fastestSolve: ScoreMap;
};

File: backend/src/main/java/de/neuefische/paulkreft/backend/user/model/Statistics.java
package de.neuefische.paulkreft.backend.user.model;

public record Statistics(
        ScoreMap longestWinningStreak,
        ScoreMap longestLosingStreak,
        ScoreMap gamesPlayed,
        ScoreMap gamesWon,
        ScoreMap fastestSolve,
        ScoreMap averageTime
) {
}
